modules/electrophysiology_browser/jsx/components/electrophysiology_session_panels.js

```diff
@@ -555,16 +560,85 @@ class FilePanel extends Component {
       </Panel>
     );
   }
+
+    /**
+     * Called by download all or download annotations button to
+     * update the annotation files before downloading
+     */
+    handleClick() {
+        console.log('The link was clicked.');
+        console.log(this.state.id);
+
+        fetch(this.props.url, {
+            method: 'PUT',
+            body: {physioFileID: this.state.id},
+        }).then((response) => {
+            if (!response.ok) {
+                console.error(response.status);
+                return;
+            }
+
+            response.json().then((data) => {
+                let msgType = 'success';
+                let message = 'Your files will be downloaded in 2 seconds!';
+                this.showAlertMessage(msgType, message);
+                console.log(message);
+            });
+        }).catch((error) => {
+            console.error(error);
+
+            let msgType = 'error';
+            let message = 'Failed to update annotation files';
+            this.showAlertMessage(msgType, message);
+            console.log(message);
+        });
+    }
+
+    /**
+     * Display a success/error alert message after form submission
+     * @param {string} msgType - error/success message
+     * @param {string} message - message content
+     */
+    showAlertMessage(msgType, message) {
+        let type = 'success';
```

Comment:
For variables with different values, I think it's better to define here and assign either in the if or else case.